consul/install.sls

```diff
@@ -0,0 +1,48 @@
+{% from "consul/map.jinja" import consul, consul_service with context %}
+
+# At the time of writing, Hashicorp doesn't maintain any official package
+# repositories for common linux distributions. The recommended installation
+# method is to download a zip, verify it, and extract consul into the system
+# path.
+
+download_package:
+  file.managed:
+    - name: /tmp/consul.zip
+    - source: {{ consul.consul_zip }}
+    - source_hash: {{ consul.consul_zip_checksum }}
```

Comment:
You can just use `archive.extracted` here. Refer to https://github.com/mitodl/vault-formula/blob/master/vault/init.sls#L8-L21 for an example.